### ISSUE TITLE
strict rule erroring for scripts when module ecma feature enabled

### ISSUE THREAD

**Issue by briandipalma** [eslint/eslint#4434]
ESLint version: 1.9.0

Using the configuration.

```javascript
{
	"rules": {
		"strict": 2,
	},
	"ecmaFeatures": {
		"modules": true
	}
}
```

Running on this script.

```javascript
"use strict"

const gretting = "Hello";
```

I expect this script (in reality the script that is triggering an error is a `webpack.config.js` file) to not cause any errors to be logged. Instead I receive,

`error  "use strict" is unnecessary inside of modules  strict`.

I was wondering if the `strict` rule could be changed to check for the module feature and for import/export statements. I realize that you can't be certain if a file is a module or not by syntactic analysis but if a file doesn't have any import/exports it seems likely that it isn't a module?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Modules are in strict mode by default. You need to define if your code is a module upfront, as there are different semantics. It's not just the presence of `import` or `export` that are the difference. When modules are natively supported in the browser and Node, you'll have to indicate you're loading a module upfront as well.

**Comment by briandipalma** [comment]
Yep, I'm aware of that. This was a question about making it more convenient for people who are using ES modules for most of their code apart from the few node tooling CJS modules. Once node added support for ES modules you would rewrite your `module.exports` etc code to use import/export form which would then cause the triggering the strict error.

Maybe this could be a config option?

**Comment by nzakas** [comment]
I'm sorry, I don't want to provide incorrect information to users. You can disable modules if you want the behavior of non-module.
